### DISCUSSION TITLE
Updating from 4.2.5 to 4.2.6 causes cluster recreation

### DISCUSSION THREAD

**Discussion by 12345ieee** [oracle-terraform-modules/terraform-oci-oke#571] [General]
This might be an issue of the terraform provider or even a OCI-wide issue, but I hope the folks here can shed some light.

Version 4.2.6 introduced native pod networking and with them this block:
https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/ffe6524ef3ad4f513bafb73c54920e1c4f3afe11/modules/oke/cluster.tf#L19-L21
in the cluster resource.

We have many clusters created before this change, which don't have that block at all in the terraform state, and when we apply with the new version we get:
```tf
 # module.oke.module.oke.oci_containerengine_cluster.k8s_cluster must be replaced
-/+ resource "oci_containerengine_cluster" "k8s_cluster" {
      [...]

      + cluster_pod_network_options { # forces replacement
          + cni_type = "FLANNEL_OVERLAY" # forces replacement
        }
```
which is really bad as it leads to whole cluster recreation even though we didn't want any changes to the network type.

Is this something I should bring to the provider? Or maybe it's something solved by updating kube to 1.24+?

**Comment by aibarbetta** [comment]
+1 opened an issue for this [here](https://github.com/oracle-terraform-modules/terraform-oci-oke/issues/591)
